src/core/java/app/com/linkedin/playparseq/j/PlayParSeqImpl.java

```diff
@@ -79,6 +80,25 @@ public PlayParSeqImpl(final Engine engine, final ParSeqTaskStore parSeqTaskStore
     });
   }
 
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public <T> Task<T> toTask(String name, HttpExecutionContext executionContext, Callable<CompletionStage<T>> f) {
+    // Bind a Task to the CompletionStage for both success and failure
+    return Task.async(name, () -> {
+      SettablePromise<T> promise = Promises.settable();
+      f.call().whenCompleteAsync((result, exception) -> {
+        if (exception != null) {
+          promise.fail(exception);
+        } else {
+          promise.done(result);
+        }
+      }, executionContext.current());
+      return promise;
+    });
+  }
```

Comment:
Could you consolidate this new method with the above? So we don't rely on copying code.

Reply:
Great call.